cwms-data-api/src/main/java/cwms/cda/api/BasinController.java

```diff
@@ -144,37 +178,149 @@ public void getOne(@NotNull Context ctx, @NotNull String basinId) {
-                NamedPgJsonFormatter basinPgJsonFormatter = new NamedPgJsonFormatter();
-                String result = basinPgJsonFormatter.format(basin);
-                ctx.result(result).contentType(contentType.toString());
+                BasinDao basinDao = new BasinDao(dsl);
+                Basin basin = basinDao.getBasin(basinName, units, office);
+                result = Formats.format(contentType, Collections.singletonList(basin), Basin.class);
+                ctx.result(result);
+                ctx.status(HttpServletResponse.SC_OK);
             } else {
-                ctx.status(HttpServletResponse.SC_NOT_FOUND).json(new CdaError("Unsupported "
-                        + "format for basins"));
+                cwms.cda.data.dao.basin.BasinDao basinDao = new cwms.cda.data.dao.basin.BasinDao(dsl);
+                CwmsId basinId = new CwmsId.Builder()
+                        .withName(basinName)
+                        .withOfficeId(office)
+                        .build();
+                cwms.cda.data.dto.basin.Basin basin = basinDao.getBasin(basinId, units);
+                result = Formats.format(contentType, Collections.singletonList(basin),
+                        cwms.cda.data.dto.basin.Basin.class);
+                ctx.result(result);
+                ctx.status(HttpServletResponse.SC_OK);
             }
         } catch (SQLException ex) {
-            String errorMsg = "Error retrieving " + basinId;
+            CdaError error = new CdaError("Error retrieving " + basinName);
+            String errorMsg = "Error retrieving " + basinName;
             LOGGER.log(Level.SEVERE, errorMsg, ex);
+            ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(error);
         }
     }
 
-    @OpenApi(ignore = true)
+    @OpenApi(
+        queryParams = {
+            @OpenApiParam(name = NAME, required = true, description = "Specifies the new name for the basin.")
+        },
+        responses = {
+            @OpenApiResponse(status = STATUS_404, description = "The provided combination of "
+                    + "parameters did not find a basin."),
+            @OpenApiResponse(status = STATUS_501, description = "Requested format is not "
+                    + "implemented")
+        },
+        description = "Renames CWMS Basin",
+        tags = {TAG}
+    )
     @Override
-    public void update(@NotNull Context ctx, @NotNull String s) {
-        ctx.status(HttpCode.NOT_IMPLEMENTED).json(CdaError.notImplemented());
+    public void update(@NotNull Context ctx, @NotNull String basinId) {
+        DSLContext dsl = getDslContext(ctx);
+        String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) :
+                Formats.NAMED_PGJSON;
+        ContentType contentType = Formats.parseHeader(formatHeader, Basin.class);
+        ctx.contentType(contentType.toString());
+
+        cwms.cda.data.dto.basin.Basin basin = Formats.parseContent(contentType, ctx.body(),
```

Comment:
Use the query parameter instead of body as per @rma-bryson 's message

Reply:
Fixed to correctly gather input from parameters. Still gets old Basin to rename from body, takes new name as parameter.